### DISCUSSION TITLE
Is it possible to filter the files in diff for their extensions?

### DISCUSSION THREAD

**Discussion by manoj798** [community/community#25991] [Actions]
<p>Is it possible to filter the files in diff for their extensions?<br>
Say for example</p>
<p>I have test1.json, test1.txt in a folder , I want to filter only test1.json using diff, I saw some answer in internet git diff – ‘*.json’ , but that didn’t work for me.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You can pass a list of files for <code>git diff</code> to compare against the current HEAD like this:</p>
<pre><code class="lang-auto">git diff -- FILE1 FILE2 ...
</code></pre>
<p>The file list can be created by a shell glob, that’s what</p>
<pre><code class="lang-auto">git diff -- *.json
</code></pre>
<p>does. However, <code>*.json</code> matches only files in the current directory. If your files are in subdirectories you have two options:</p>
<ol>
<li>Use <code>find</code> to build the file list, e.g.</li>
</ol>
<pre><code class="lang-auto">git diff -- $(find . -name *.json)
</code></pre>
<ol start="2">
<li>If you’re using bash you can set the <code>globstar</code> shell option and use <code>**</code> to match any number of directories:</li>
</ol>
<pre><code class="lang-auto">shopt -s globstar
git diff -- **/*.json
</code></pre>
